variables.tf

```diff
@@ -155,19 +214,32 @@ variable "cur_report_frequency" {
     condition     = can(regex("^(DAILY|HOURLY|MONTHLY|NONE)$", var.cur_report_frequency))
     error_message = "Valid options: DAILY, HOURLY, MONTHLY, NONE"
   }
-
 }
 
 #
 # SCPs & OUs
 variable "service_control_policies" {
   description = "Map of SCPs to deploy"
   default     = {}
+  type = map(
+    object({
+      policy_name        = string
+      policy_description = string
+      policy_json_file   = string
+      policy_targets     = optional(list(string))
+    })
+  )
 }
 
 variable "organization_units" {
   description = "Map of OUs to deploy"
   default     = {}
+  type = map(
+    object({
+      name             = string
```

Comment:
```suggestion
      parent_id             = optional(string)
```

Reply:
Fixing, but it should be both `name` and `parent_id`. The resource is defined as:

```hcl
resource "aws_organizations_organizational_unit" "custom_ous" {
  for_each  = var.organization_units
  name      = each.value["name"]
  parent_id = each.value["is_child_of_root"] ? aws_organizations_organization.org.roots[0].id : local.ou_name_to_id[each.value["parent_id"]]
}
```

Based on that change I will add validation so `parent_id` is only set if `is_child_of_root` is `false`. I will also make `is_child_of_root` default to `true`.

Reply:
While implementing the change, I realised `is_child_of_root` is redundant. We can use `parent_id = optional(string)` and if it is `null` use the root as the parent. I have retained `parent_id = optional(bool)` for backwards compatibility.

Reply:
Makes sense. I probably did that when my terraform-fu was weak.